samza-core/src/main/java/org/apache/samza/clustermanager/ResourceRequestState.java

```diff
@@ -69,6 +69,18 @@ public ResourceRequestState(boolean hostAffinityEnabled, ClusterResourceManager
     this.manager = manager;
   }
 
+  public void cancelResourceRequest(SamzaResourceRequest request) {
```

Comment:
Minor: Move lower (to after updateStateAfterAssignment) and maybe add documentation.